internal/dataplane/deckerrors/api.go

```diff
@@ -18,13 +20,26 @@ func ExtractAPIErrors(err error) []*kong.APIError {
 	// It might be either a deckutils.ErrArray with APIErrors inside.
 	var deckErrArray deckutils.ErrArray
 	if errors.As(err, &deckErrArray) {
-		var apiErrs []*kong.APIError
-		for _, err := range deckErrArray.Errors {
-			if apiErr, ok := castAsErr[*kong.APIError](err); ok {
-				apiErrs = append(apiErrs, apiErr)
-			}
-		}
-		return apiErrs
+		return lo.FilterMap(deckErrArray.Errors, func(e error, _ int) (*kong.APIError, bool) {
+			return castAsErr[*kong.APIError](e)
+		})
+	}
+
+	return nil
+}
+
+func ExtractCRUDActionErrors(err error) []*crud.ActionError {
```

Comment:
Please consider covering this function with a unit test similar to `TestExtractAPIErrors`.

Reply:
Added unit tests.